rules/contrib/views.py

```diff
@@ -49,6 +50,84 @@ def has_permission(self):
         return self.request.user.has_perms(perms, obj)
 
 
+class AutoPermissionRequiredMixin(PermissionRequiredMixin):
+    """
+    An extended variant of PermissionRequiredMixin which automatically determines
+    the permission to check based on the type of view it's used with.
+
+    It works by checking the current view for being an instance of a pre-defined
+    list of view types. On a match, the corresponding permission type (such as
+    "add" or "change") is converted into the full model-specific permission name
+    and checked. See the permission_type_map attribute for the default view type ->
+    permission type mappings.
+
+    When a view using this mixin has an attribute ``permission_type``, that type is
+    used directly and overwrites the permission_type_map for the particular view. A
+    permission type of ``None`` causes permission checking to be skipped. If the
+    type of permission to check for should depend on dynamic factors other than the
+    view type, you may overload the ``permission_type`` attribute with a ``property``.
```

Comment:
“``property``” -> “*property*” or just “property” (i.e. not formatted as code)

Reply:
I'm referencing to Pythons `property` decorator, hence I think code formatting is appropriate here.

Reply:
Maybe `@property` then? Otherwise, I don't think I agree here -- the sentence seems to be referring to the concept/semantics of a property, which just happens to be created with the `@property` decorator in Python. If the decorator was named `makeProperty`, would you write the sentence as "overload the ``permission_type`` attribute with a ``makeProperty``"? :)

BTW, I think "overload" should be "override" here too.

Reply:
After wrapping with `@property`, the class member is an instance of the `property` class. But anyway, I'm going to change it.